test/e2e/e2e.go

```diff
@@ -0,0 +1,198 @@
+		initOptions := []string{"--domain", c.domain}
+		err := kubebuilderTest.Init(initOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating resource definition")
+		resourceOptions := []string{"--group", c.group, "--version", c.version, "--kind", c.kind}
+		err = kubebuilderTest.CreateResource(resourceOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating core-type resource controller")
+		controllerOptions := []string{"--group", "apps", "--version", "v1beta2", "--kind", "Deployment", "--core-type"}
+		err = kubebuilderTest.CreateController(controllerOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("building image")
+		// The scaffold test cases generated for core types cannot work without manually modification.
+		// See https://github.com/kubernetes-sigs/kubebuilder/pull/193 for more details
+		// Skip the test part in build process as we don't care about it here.
+		err = framework.ReplaceFileConent(`RUN go test(.*)\n`, "", filepath.Join(c.workDir, "Dockerfile.controller"))
+		Expect(err).NotTo(HaveOccurred())
+
+		imageOptions := []string{"-t", c.controllerImageName}
+		err = kubebuilderTest.BuildImage(imageOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating config")
+		configOptions := []string{"--controller-image", c.controllerImageName, "--name", c.installName}
+		err = kubebuilderTest.CreateConfig(configOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("installing controller-manager in cluster")
+		inputFile := filepath.Join(kubebuilderTest.Dir, "hack", "install.yaml")
+		installOptions := []string{"apply", "-f", inputFile}
+		_, err = kubebuilderTest.RunKubectlCommand(framework.GetKubectlArgs(installOptions))
+		Expect(err).NotTo(HaveOccurred())
+
+		By("validate the controller-manager pod running as expected")
+		verifyContollerUp := func() (bool, error) {
+			e = nil
+			// Get pod name
+			getOptions := []string{"get", "pods", "-n", c.namespace, "-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}{{ \"\\n\" }}{{ end }}{{ end }}"}
+			podOutput, err := kubebuilderTest.RunKubectlCommand(framework.GetKubectlArgs(getOptions))
+			Expect(err).NotTo(HaveOccurred())
+			// TODO: validate pod replicas if not default to 1
+			podNames := framework.ParseCmdOutput(podOutput)
+			if len(podNames) != 1 {
+				e = fmt.Errorf("expect 1 controller pods running, but got %d", len(podNames))
+				return false, nil
+			}
+			controllerPodName = podNames[0]
+
+			// Validate pod status
+			getOptions = []string{"get", "pods", controllerPodName, "-n", c.namespace, "-o", "jsonpath={.status.phase}"}
+			status, err := kubebuilderTest.RunKubectlCommand(framework.GetKubectlArgs(getOptions))
+			Expect(err).NotTo(HaveOccurred())
+			if status != "Running" {
+				e = fmt.Errorf("controller pod in %s status", status)
+				return false, nil
+			}
+
+			return true, nil
+		}
+		varifyErr := wait.PollImmediate(500*time.Millisecond, 1*time.Minute, verifyContollerUp)
```

Comment:
why not use `Eventually` from ginkgo?

Reply:
https://onsi.github.io/gomega/#eventually

Reply:
done